src/controls/color-picker.js

```diff
@@ -35,46 +36,60 @@ const propTypes = {
   onOpen: PropTypes.func.isRequired,
   onClose: PropTypes.func.isRequired,
   active: PropTypes.bool,
+  containerRef: PropTypes.shape({ current: PropTypes.any }),
 }
 
 const defaultProps = {
   value: '#000000',
   onChange: noop,
   onOpen: noop,
   onClose: noop,
+  containerRef: null,
 }
 
-function ColorPicker({ active, value, onChange, onOpen, onClose, ...rest }) {
+function ColorPicker({
+  active,
+  value,
+  onChange,
+  onOpen,
+  onClose,
+  containerRef,
+  ...rest
+}) {
   const [expanded, toggleExpanded] = useToggle()
   const isExpanded = active === undefined ? expanded : active
 
   return (
     <div className="color-picker">
-      <span
-        className="swatch"
-        style={{ background: value || 'black' }}
-        onClick={() => {
-          toggleExpanded()
-          return onOpen()
+      <OutsideClickHandler
+        onOutsideClick={(e) => {
+          if (containerRef && containerRef.current.contains(e.target)) {
```

Comment:
This allows the `ColorInput` to define its own target area for what it considers to be "outside". When clicking on the input, we want the chrome picker to stay open, as opposed to getting closed by `toggleExpanded(false)`